src/chapter02/exercises/exercise03.lean

```diff
@@ -7,13 +7,26 @@ def rational (x : ℝ) :=
 /-- The product of two rational numbers is always rational. -/
 lemma part_a : ∀ {a b : ℝ}, rational a → rational b → rational (a * b) :=
 begin
-  sorry
+  rintro a b ⟨x1,y1,rfl⟩ ⟨x2,y2,rfl⟩,
+  use [x1*x2, y1*y2],
+  push_cast,
+  simp only [div_eq_mul_inv, mul_inv],
+  ring,
 end
 
 /-- The product of two irrational numbers is not always irrational. -/
 lemma part_b : ¬ ∀ {a b : ℝ}, irrational a → irrational b → irrational (a * b) :=
 begin
-  sorry
+  push_neg,
+  use [real.sqrt 2, real.sqrt 2],
+  split,
+  { exact irrational_sqrt_two },
+  { split,
```

Comment:
You can do all of `use` and `split` and `exact` with the `refine` tactic. Try this:
```
  refine ⟨real.sqrt 2, real.sqrt 2, irrational_sqrt_two, irrational_sqrt_two, _⟩,
```
on line 21, and that can be used to replace the first five lines of your proof.